### PR TITLE
Usecase2

### PR DESCRIPTION
**Pull request by jariyarathne**
This PR contains the changes required for Use Case 2 - integration of client and back-end with data migration services.

### CODE REVIEW

File: manage_employees/employee_management_services.bal [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        EmployeeInfo employeeInfo;
+        http:Request hrSystemRequest = new;
+        json requestJson = {};
+        string employeeId;
+        string newEmployeeId;
+        string departmentId;
+        string newDepartmentId;
+
+        //Process request JSON payload and validate
+        var payloadJson = addEmployeeRequest.getJsonPayload();
```
Comment (saadsahibjan) [inline manage_employees/employee_management_services.bal]: without assigning the json payload to a variable, why don't you have it as an argument for match.

match (addEmployeeRequest.getJsonPayload()) { ... }

File: hr_system/hr_employee_mgt_services.bal [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+}
+service<http:Service> hrEmployeeManagementService bind EmployeeMgtServiceListener {
+
+    @http:ResourceConfig {
+        methods: ["POST"],
+        path: "/employee"
+    }
+    addEmployee(endpoint caller, http:Request addEmployeeReq) {
+        InputData inputData;
+        Employee employee;
```
Comment (saadsahibjan) [inline hr_system/hr_employee_mgt_services.bal]: Any reasons to declare these two objects here and assigning values in few lines after?

File: manage_employees/employee_management_services.bal (hunk 1 above)
Comment (jariyarathne) [reply manage_employees/employee_management_services.bal]: Fixed

File: hr_system/hr_employee_mgt_services.bal (hunk 2 above)
Comment (jariyarathne) [reply hr_system/hr_employee_mgt_services.bal]: Fixed

**Pull request merged**